tests/utils/cms/cms.go

```diff
@@ -0,0 +1,247 @@
+	if len(params) == 1 {
+		for k, v := range params[0] {
+			request = request.Parameter(k, v)
+		}
+	}
+	return request.Send()
+}
+
+// RetrieveMachineTypeDetail will return the retrieve result of machine type detailed information
+func RetrieveMachineTypeDetail(connection *client.Connection, machineTypeID string) (*client.Response, error) {
+	return connection.Get().Path(fmt.Sprintf(machineTypeIDURL, machineTypeID)).Send()
+}
+
+func RetrieveIDP(connection *client.Connection, clusterID string, idpID string) (*cmv1.IdentityProviderGetResponse, error) {
+	return connection.ClustersMgmt().V1().Clusters().Cluster(clusterID).IdentityProviders().IdentityProvider(idpID).Get().Send()
+}
+
+func ListIDPs(connection *client.Connection, clusterID string) (*cmv1.IdentityProvidersListResponse, error) {
+	return connection.ClustersMgmt().V1().Clusters().Cluster(clusterID).IdentityProviders().List().Send()
+}
+
+// RetrieveClusterCPUTotalByNodeRolesOS will return the physical cpu_total of the compute nodes of the cluster
+func RetrieveClusterCPUTotalByNodeRolesOS(connection *client.Connection, clusterID string) (*cmv1.CPUTotalByNodeRolesOSMetricQueryGetResponse, error) {
+	return connection.ClustersMgmt().V1().Clusters().Cluster(clusterID).MetricQueries().CPUTotalByNodeRolesOS().Get().Send()
+}
+
+// RetrieveClusterSocketTotalByNodeRolesOS will return the physical socket_total of the compute nodes of the cluster
+func RetrieveClusterSocketTotalByNodeRolesOS(connection *client.Connection, clusterID string) (*cmv1.SocketTotalByNodeRolesOSMetricQueryGetResponse, error) {
+	return connection.ClustersMgmt().V1().Clusters().Cluster(clusterID).MetricQueries().SocketTotalByNodeRolesOS().Get().Send()
+}
+
+func RetrieveDetailedIngressOfCluster(connection *client.Connection, clusterID string, ingressID string) (*cmv1.IngressGetResponse, error) {
+	resp, err := connection.ClustersMgmt().V1().Clusters().Cluster(clusterID).Ingresses().Ingress(ingressID).Get().Send()
+	return resp, err
+}
+
+// Cluster labels
+func ListClusterExternalConfiguration(connection *client.Connection, clusterID string) (*cmv1.ExternalConfigurationGetResponse, error) {
+	resp, err := connection.ClustersMgmt().V1().Clusters().Cluster(clusterID).ExternalConfiguration().Get().Send()
+	return resp, err
+}
+
+func ListClusterLabels(connection *client.Connection, clusterID string, parameter ...map[string]interface{}) (*cmv1.LabelsListResponse, error) {
+	request := connection.ClustersMgmt().V1().Clusters().Cluster(clusterID).ExternalConfiguration().Labels().List()
+	for _, param := range parameter {
+		for k, v := range param {
+			request = request.Parameter(k, v)
+		}
+	}
+	return request.Send()
+}
+
+func RetrieveDetailedLabelOfCluster(connection *client.Connection, clusterID string, labelID string) (*cmv1.LabelGetResponse, error) {
+	resp, err := connection.ClustersMgmt().V1().Clusters().Cluster(clusterID).ExternalConfiguration().Labels().Label(labelID).Get().Send()
+	return resp, err
```

Comment:
```suggestion
	return connection.ClustersMgmt().V1().Clusters().Cluster(clusterID).ExternalConfiguration().Labels().Label(labelID).Get().Send()
```

Reply:
@amalykhi I see this is marked as resolved, but not really in the code
same goes for commented-out code and other stuff
is it that you didn't yet committed the changes?

Reply:
I accepted change in github ui, but looks it didn't work properly. Fixed it